controller/src/beerocks/master/tasks/dynamic_channel_selection_task.cpp

```diff
@@ -252,6 +252,15 @@ void dynamic_channel_selection_task::handle_event(int event_type, void *obj)
     }
     case eEvent::SCAN_TRIGGER_FAILED: {
         TASK_LOG(DEBUG) << "SCAN_TRIGGER_FAILED received";
+        if (fsm_in_state(eState::WAIT_FOR_SCAN_TRIGGERED)) {
+            auto scan_trigger_failed_event = reinterpret_cast<sScanEvent *>(obj);
```

Comment:
Is there a chance that we'll received a `SCAN_TRIGGER_FAILED` event not from the radio we're waiting for?
You have `radio_mac` in the event, but you don't check it...